Add a new extension to search confluence wiki

daviddkkim opened the pull request:
## Description

This is a new extension that allows users to search Confluence Wiki.
They would authenticate by specifying their domain, username and api token. Afterwards, they can search pages as if they are on the browser. 

You can either navigate to the page or copy URL to clipboard.

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/12561526/149828901-04b39c07-53b2-499e-88a7-cff25fdf8f5f.mp4


## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

pernielsentikaer commented on extensions/confluence-search/package.json:
```diff
@@ -0,0 +1,64 @@
+  "commands": [
+    {
+      "name": "index",
+      "title": "Search Confluence Wiki",
```
```suggestion
      "title": "Search Wiki",
      "subtitle": "Confluence",
```

pernielsentikaer reviewed (changes requested):
Hi @daviddkkim 👋 

Thanks for your contribution 💯 

I don't have credentials to test it fully, do you by any chance have some credentials I could borrow a few minutes?

One thing I noticed is that when wrong credentials are present, it doesn't die gracefully (showing a toast and stopping the loader with an empty array) – Can you look into this?

![v6zs1oQBhRdViP1J](https://user-images.githubusercontent.com/38033107/150071849-74a6079c-4cbc-4e27-88b2-640a261d72a5.png)

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

daviddkkim commented:
Screen capture including the authentication

https://user-images.githubusercontent.com/12561526/150695809-0132ed52-5380-4910-81c3-1631c51a46bc.mp4

daviddkkim commented:
@pernielsentikaer 

Made changes based on our slack convo:
- removed unecessary await 
- added in proper loading state based on whether the API call has returned rather than list item length
- changed pref field name from domain url to domain.


The empty list is there to show empty list in case API call response status was not OK. (fail state).

pernielsentikaer commented:
I sent you some more feedback in our conversation in Slack, let me know if you need anything 😄

daviddkkim commented:
@pernielsentikaer changed made !

pernielsentikaer commented on extensions/confluence-search/src/index.tsx:
```diff
@@ -0,0 +1,205 @@
+  );
+}
+
+enum Status {
+  Failure,
+  Success,
+}
+
```
```suggestion
```
Unused enum status

pernielsentikaer commented on extensions/confluence-search/src/index.tsx:
```diff
@@ -0,0 +1,205 @@
+      accessoryIcon={{ source: `${confluenceUrl}${searchResult.icon}`, mask: ImageMask.Circle }}
+      icon={Icon.Document}
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section>
+            <OpenInBrowserAction title="Open in Browser" url={confluenceUrl + "/wiki" + searchResult.url} />
+          </ActionPanel.Section>
+          <ActionPanel.Section>
+            <CopyToClipboardAction
+              title="Copy URL"
+              content={confluenceUrl + "/wiki" + searchResult.url}
+              shortcut={{ modifiers: ["cmd"], key: "." }}
+            />
+          </ActionPanel.Section>
+        </ActionPanel>
```
```suggestion
        <ActionPanel>
          <ActionPanel.Section>
            <OpenInBrowserAction title="Open in Browser" url={confluenceUrl + "/wiki" + searchResult.url} />
            <CopyToClipboardAction
              title="Copy URL"
              content={confluenceUrl + "/wiki" + searchResult.url}
              shortcut={{ modifiers: ["cmd"], key: "." }}
            />
          </ActionPanel.Section>
        </ActionPanel>
```
You could put it in the same section

pernielsentikaer reviewed (changes requested):
Hi @daviddkkim 👋 

Awesome, thanks for the corrections 💯  Only two small NITs back and we're ready to proceed 💪

daviddkkim commented:
> Hi @daviddkkim 👋
> 
> Awesome, thanks for the corrections 💯 Only two small NITs back and we're ready to proceed 💪

Thanks Per!

pernielsentikaer reviewed (approved):
Awesome @daviddkkim 💪 

Thanks for your contribution 💯 I'm going to accept it and then will my colleague do a final review before it's finally merged.

sxn commented:
🎉
